mmv1/third_party/terraform/services/resourcemanager/ephemeral_google_service_account_id_token_test.go

```diff
@@ -0,0 +1,107 @@
+package resourcemanager_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+	"github.com/hashicorp/terraform-provider-google/google/envvar"
+)
+
+func TestAccEphemeralServiceAccountIdToken_basic(t *testing.T) {
+	t.Parallel()
+
+	serviceAccount := envvar.GetTestServiceAccountFromEnv(t)
+	targetServiceAccountEmail := acctest.BootstrapServiceAccount(t, "idtoken", serviceAccount)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		ExternalProviders: map[string]resource.ExternalProvider{
+			"time": {},
+		},
+		Steps: []resource.TestStep{
+			{
+				Config: testAccEphemeralServiceAccountIdToken_basic(targetServiceAccountEmail),
+			},
+		},
+	})
+}
+
+func TestAccEphemeralServiceAccountIdToken_withDelegates(t *testing.T) {
+	t.Parallel()
+
+	initialServiceAccount := envvar.GetTestServiceAccountFromEnv(t)
+	delegateServiceAccountEmailOne := acctest.BootstrapServiceAccount(t, "delegate1", initialServiceAccount)          // SA_2
+	delegateServiceAccountEmailTwo := acctest.BootstrapServiceAccount(t, "delegate2", delegateServiceAccountEmailOne) // SA_3
+	targetServiceAccountEmail := acctest.BootstrapServiceAccount(t, "target", delegateServiceAccountEmailTwo)         // SA_4
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		ExternalProviders: map[string]resource.ExternalProvider{
+			"time": {},
+		},
+		Steps: []resource.TestStep{
+			{
+				Config: testAccEphemeralServiceAccountIdToken_withDelegates(delegateServiceAccountEmailOne, delegateServiceAccountEmailTwo, targetServiceAccountEmail),
+			},
+		},
+	})
+}
+
+func TestAccEphemeralServiceAccountIdToken_withIncludeEmail(t *testing.T) {
```

Comment:
Once the edit to GetCredentials' arguments is included, this test fails due to:

```
ephemeral_google_service_account_id_token_test.go:62: Step 1/1 error: Error running pre-apply refresh: exit status 1
        
        Error: Error calling iamcredentials.GenerateIdToken
        
          with ephemeral.google_service_account_id_token.token,
          on terraform_plugin_test.tf line 3, in ephemeral "google_service_account_id_token" "token":
           3: ephemeral "google_service_account_id_token" "token" {
        
        googleapi: Error 403: Permission 'iam.serviceAccounts.getOpenIdToken' denied
        on resource (or it may not exist).
        Details:
        [
          {
            "@type": "type.googleapis.com/google.rpc.ErrorInfo",
            "domain": "iam.googleapis.com",
            "metadata": {
              "permission": "iam.serviceAccounts.getOpenIdToken"
            },
            "reason": "IAM_PERMISSION_DENIED"
          }
        ]
        , forbidden

```

[This code](https://github.com/GoogleCloudPlatform/magic-modules/blob/8b9073c0504d25f6978af4d8b4f5e912d9bc9269/mmv1/third_party/terraform/services/resourcemanager/ephemeral_google_service_account_id_token_test.go#L57-L58) bootstraps the existence of both service accounts and makes sure that `serviceAccount` has permissions to imitate `targetServiceAccountEmail` via roles/iam.serviceAccountTokenCreator. That's insufficient for this test though, hence the error.

Please add a step in the test where the missing permission is created. Then the ephemeral resource can be added in the second step.

Reply:
I don't think this is necessary, the bootstrap serviceaccount should already do this since it's performing the same setup as all the other tests where each one also requires the `iam.serviceAccountTokenCreator`:
```hcl
	serviceAccount := envvar.GetTestServiceAccountFromEnv(t)
	targetServiceAccountEmail := acctest.BootstrapServiceAccount(t, "idtoken-email", serviceAccount)
```

I ran the test and got the same error but waited a bit for the permissions to be setup, afterwards I got passing. The same was necessary for all other tests in this PR. When running them all again today.

```hcl
└─(15:05:51 on main ✹ ✭)──> envchain GCLOUD make testacc TESTARGS='-run=TestAccEphemeralServiceAccountIdToken_withIncludeEmail' TEST=./google/services/resourcemanager/...
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google/services/resourcemanager/... -v -run=TestAccEphemeralServiceAccountIdToken_withIncludeEmail -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google/version.ProviderVersion=acc"
=== RUN   TestAccEphemeralServiceAccountIdToken_withIncludeEmail
=== PAUSE TestAccEphemeralServiceAccountIdToken_withIncludeEmail
=== CONT  TestAccEphemeralServiceAccountIdToken_withIncludeEmail
2024/11/18 15:07:35 [INFO] Authenticating using configured Google JSON 'credentials'...
2024/11/18 15:07:35 [INFO]   -- Scopes: [https://www.googleapis.com/auth/cloud-platform https://www.googleapis.com/auth/userinfo.email]
2024/11/18 15:07:35 [INFO] Authenticating using configured Google JSON 'credentials'...
2024/11/18 15:07:35 [INFO]   -- Scopes: [https://www.googleapis.com/auth/cloud-platform https://www.googleapis.com/auth/userinfo.email]
2024/11/18 15:07:35 [DEBUG] Waiting for state to become: [success]
2024/11/18 15:07:36 [INFO] Terraform is using this identity: mauricio-alvarezleon@hc-terraform-testing.iam.gserviceaccount.com
2024/11/18 15:07:36 [DEBUG] Verifying projects/hc-terraform-testing/serviceAccounts/tf-bootstrap-sa-idtoken-email@hc-terraform-testing.iam.gserviceaccount.com as bootstrapped service account.
2024/11/18 15:07:36 [INFO] Instantiating Google Cloud IAM client for path https://iam.googleapis.com/
2024/11/18 15:07:36 [DEBUG] Retry Transport: starting RoundTrip retry loop
2024/11/18 15:07:36 [DEBUG] Retry Transport: request attempt 0
2024/11/18 15:07:36 [DEBUG] Retry Transport: Stopping retries, last request was successful
2024/11/18 15:07:36 [DEBUG] Retry Transport: Returning after 1 attempts
2024/11/18 15:07:36 [DEBUG] Setting service account permissions.
2024/11/18 15:07:36 [INFO] Instantiating Google Cloud IAM client for path https://iam.googleapis.com/
2024/11/18 15:07:36 [DEBUG] Retry Transport: starting RoundTrip retry loop
2024/11/18 15:07:36 [DEBUG] Retry Transport: request attempt 0
2024/11/18 15:07:37 [DEBUG] Retry Transport: Stopping retries, last request was successful
2024/11/18 15:07:37 [DEBUG] Retry Transport: Returning after 1 attempts
--- PASS: TestAccEphemeralServiceAccountIdToken_withIncludeEmail (20.21s)
PASS
ok      github.com/hashicorp/terraform-provider-google/google/services/resourcemanager  21.826s
```